### PR TITLE
fix server yaml and failure log

### PR DESCRIPTION
**Pull request by mandarjog**
In case of evaluation failure, log the expression that failed to eval.

### CODE REVIEW

File: extensions/attributegen/testdata/server.yaml [hunk 1]
```diff
@@ -104,14 +104,14 @@ static_resources:
...
+              type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
               value:
                 config:
                   configuration: |
-                    {
+                    { "debug": "true",
                       "attributes": [
                         {
-                          "output_attribute": "istio.operationId",
+                          "output_attribute": "istio_operationId",
```
Comment (douglas-reid) [inline extensions/attributegen/testdata/server.yaml:114]: is there a separate PR for fixing the docs re: namespacing of attributes?

File: extensions/stats/plugin.cc [hunk 2]
```diff
@@ -587,8 +587,15 @@ bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
...
+      std::string expression_str;
+      for (const auto& expression : input_expressions_) {
+        if (expression.second == i) {
+          expression_str = expression.first;
+          break;
+        }
+      }
+      LOG_TRACE(absl::StrCat("Failed to evaluate expression: <", expression_str,
+                             "> at slot: ", std::to_string(i)));
       istio_dimensions_[count_standard_labels + i] = "";
```
Comment (douglas-reid) [inline extensions/stats/plugin.cc:599]: i'm beginning to wonder if `""` is the best value here. didn't we have something elsewhere in this that was auto-converting `""` to `"unknown"` ? are we now inconsistent in missing/errors for dimensions?

File: extensions/stats/plugin.cc [hunk 3]
```diff
@@ -587,8 +587,15 @@ bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
...
-      LOG_TRACE(absl::StrCat("Failed to evaluate expression at slot: " +
-                             std::to_string(i)));
+      std::string expression_str;
+      for (const auto& expression : input_expressions_) {
+        if (expression.second == i) {
+          expression_str = expression.first;
+          break;
+        }
+      }
+      LOG_TRACE(absl::StrCat("Failed to evaluate expression: <", expression_str,
```
Comment (douglas-reid) [inline extensions/stats/plugin.cc:597]: is the `slot` information useful? i wouldn't know what to do with it.  is there a way to say which dimension the expression was intended to populate ?

Comment (mandarjog) [reply extensions/stats/plugin.cc]: Updated, adding dimension is a pain, i will take a fresh look tomorrow.

File: extensions/stats/plugin.cc (hunk 2 above)
Comment (mandarjog) [reply extensions/stats/plugin.cc:596]: do we want to change this to unknown or unmapped?

File: extensions/attributegen/testdata/server.yaml (hunk 1 above)
Comment (mandarjog) [reply extensions/attributegen/testdata/server.yaml:114]: Will send another PR.

**Comment by mandarjog** [comment]
@bianpengyuan updated to align with attributegen that keep token and expression in a struct.

File: extensions/stats/plugin.cc (hunk 2 above)
Comment (bianpengyuan) [reply extensions/stats/plugin.cc:596]: `unknown` sounds better to me. It will be a good indicator that config takes effect, but evaluation fails.

**Comment by bianpengyuan** [comment]
Need to update this line to unknown: https://github.com/istio/proxy/blob/147136932569f02b9959e7915ebeac55881bddca/testdata/metric/client_request_total_customized.yaml.tmpl#L23

**Pull request merged**